ethcore/src/state/mod.rs

```diff
@@ -1154,7 +1158,9 @@ impl<B: Backend> State<B> {
 
 	/// Replace account code and storage. Creates account if it does not exist.
 	pub fn patch_account(&self, a: &Address, code: Arc<Bytes>, storage: HashMap<H256, H256>) -> TrieResult<()> {
-		Ok(self.require(a, false)?.reset_code_and_storage(code, storage))
+		let mut account = self.require(a, false)?;
+		account.switch_to_patched();
+		Ok(account.reset_code_and_storage(code, storage))
```

Comment:
I got a question regarding 'reset_code_and_storage' : currently it does not touch account fields 'original_storage_cache', shouldn't we (I am not 100% sure but it seems safer) ?

In fact the PR seems to try to ignore lru storage cache since storage is ignored in private_tx, but to get eip-1283 running we probably want to keep trace of original_storage_value, I am starting to wonder how it will behave.

Reply:
My understanding of eip-1283 is gas optimization of SSTORE calls. For this case original cache may be required. In terms of this PR I fixed caching logic for SLOAD, which doesn't use original cache. 
But again it's only my understanding. I don't know, how original cache can be further evolved. So it would be great, if @sorpaas can comment on this.

Reply:
I imagine plugging the encrypted state as a replacement for the trieDB storage could make it run ok, but there may be simpler way to do it.
Unrelated question, about those private contract, what are the limitation, it seems logic that they cannot change the state of other contracts (kinda break the confidentiality), but are they allowed to call external contract?